### ISSUE TITLE
Single or isolated data point will not render any visual cue besides Tooltip.

### ISSUE THREAD

**Issue by joaovfabri** [recharts/recharts#5218]
## What is the current behavior?
Current Version of Recharts (2.12.7) still presents a bug identical to the one reported in this issue #149. 

Rendering an Area / Bar chart with isolated points (not only with single data points) will not display any representation in the chart for that point.

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).
Reproduction is the same as mentioned issue, #149

## What is the expected behavior?
 Similar libraries will represent these points with e.g. lines coming from the x axis and back: 
![Screenshot 2024-11-07 at 08 45 30](https://github.com/user-attachments/assets/2f8bda64-496e-4971-a6e6-34058e4a4fd2)

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
(2.12.7) Any browser

**Comment by ckifer** [comment]
Hey, sorry that this is still an issue. That issue was a closed a long time ago before any of the current maintainers were around.

Reproduction
https://codesandbox.io/p/sandbox/simple-line-chart-forked-tl8mw4

Looks like the intention is that a dot should be rendered if there is one datapoint. 
The problem being one data point is different than many datapoints with only 1 of them having a Y value

<img width="469" alt="image" src="https://github.com/user-attachments/assets/ab17acbc-5b8d-43f3-ab28-d841e3a87c94">

**Comment by ckifer** [comment]
https://github.com/recharts/recharts/blob/fd50fb8c67a60b5b9e8af3388afc4212396838ed/src/cartesian/Area.tsx#L513

That'll do it

**Comment by ckifer** [comment]
Don't see any issue with Bar chart 
https://codesandbox.io/p/sandbox/simple-line-chart-forked-9f449v

**Comment by joaovfabri** [comment]
Hey! that's a lightning fast response, thanks a lot for that. It can be a problem as well when a set of points for a line contains multiple Nulls and a single point in between, like 
```javascript 
{ varX: "Mon", varY: 10 },
{ varX: "Tue", varY: 8 },
{ varX: "Wed", varY: 5 },
{ varX: "Thu" },
{ varX: "Fri" },
{ varX: "Sat", varY: 10 },
{ varX: "Sun" },
```

**Comment by ckifer** [comment]
I believe if you are using the `dot` prop the dot will show if in that case. If you are not using the dot prop I'm not sure what to do there - If you want those lines connected you can use connectNulls, otherwise it's just a point (without a dot) that isn't connected to anything. 

Open to suggestions I guess

**Comment by ckifer** [comment]
![image](https://github.com/user-attachments/assets/9b50389c-4df9-44d2-8889-5da7b63bdb77)

**Comment by TechyChan** [comment]
I'm also running into this issue for my Area Chart, where for isolated data points (no data point left or right), nothing is rendered. In this case, I would prefer to have a dot rendered. Would be great to have this property configurable for Area and Line; otherwise I'll have to do a bit hacky logic in the custom dot function and check left + right of current datapoint to see if it's an isolated point.

**Comment by ckifer** [comment]
@TechyChan https://codesandbox.io/p/sandbox/simple-area-chart-forked-knxs78

There is a single dot rendered when there is a single data point. What are you looking for?

**Comment by TechyChan** [comment]
Thanks for the quick response!

Here's the behavior I'm looking for: https://codesandbox.io/p/sandbox/simple-area-chart-forked-mx8dpk

For non-isolated data points, I want a smooth line with area beneath. For isolated points, a dot should be rendered. I guess I'm able to implement this functionality since the `dot` callback has all the arguments I need, thanks!

**Comment by ckifer** [comment]
Makes sense. Since this has a workaround with the callback to the dot prop I'm going to leave things as is for now, especially in 2.x. Maybe will look into adding something else before 3.0, especially since changing any default behavior would be breaking

**Comment by Rodjxc** [comment]
In case it helps anyone in the future, I was facing the same issue and I've found a prop called connectNulls={true} that you can apply that does exactly that, connects the "lone" points with an imaginary line.
